### PR TITLE
feat: Upgrade Mobile Storybook to version 6.5

### PR DESCRIPTION
**Pull request by brianacnguyen**
## **Description**
- Upgrade Mobile Storybook to version 6.5
- Currently only uses samples story as a sample

## **Screenshots/Recordings**
Simulator Screen Shot - iPhone 14 Pro - 2023-09-21 at 11 56 33
Simulator Screen Shot - iPhone 14 Pro - 2023-09-21 at 11 56 32
Simulator Screen Shot - iPhone 14 Pro - 2023-09-21 at 11 56 30

## **Related issues**

_Fixes #???_

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -893,11 +904,12 @@ SPEC CHECKSUMS:
...
-  RNCAsyncStorage: 0c357f3156fcb16c8589ede67cc036330b6698ca
+  RNCAsyncStorage: c913ede1fa163a71cea118ed4670bbaaa4b511bb
   RNCCheckbox: ed1b4ca295475b41e7251ebae046360a703b6eb5
   RNCClipboard: ddd4d291537f1667209c9c405aaa4307297e252e
   RNCMaskedView: c298b644a10c0c142055b3ae24d83879ecb13ccd
   RNCPicker: cb57c823d5ce8d2d0b5dfb45ad97b737260dc59e
+  RNDateTimePicker: b722c23030b744763cf51d80fc22f354653f65e0
   RNDefaultPreference: 2f8d6d54230edbd78708ada8d63bb275e5a8415b
   RNDeviceInfo: 1e3f62b9ec32f7754fac60bd06b8f8a27124e7f0
   RNFS: 3ab21fa6c56d65566d1fb26c2228e2b6132e5e32
```
Comment (georgewrmarshall) [inline ios/Podfile.lock:915]: Should this file be here?

File: .storybook/storybook.requires.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  addParameters,
+  addArgsEnhancer,
+  clearDecorators,
+} from "@storybook/react-native";
+
+global.STORIES = [
+  {
+    titlePrefix: "",
+    directory: "./.storybook/stories",
+    files: "**/*.stories.?(ts|tsx|js|jsx)",
```
Comment (georgewrmarshall) [inline .storybook/storybook.requires.js:15]: Do we use any `stories.` `js`|`jsx` files? Could we just use `tsx`?

File: ios/Podfile.lock (hunk 1 above)
Comment (Cal-L) [reply ios/Podfile.lock:915]: +1 seems like this can be removed

File: package.json [hunk 3]
```diff
@@ -299,7 +301,7 @@
     "react-native-randombytes": "^3.5.3",
     "react-native-reanimated": "3.1.0",
     "react-native-redash": "16.2.2",
-    "react-native-safe-area-context": "^3.1.9",
+    "react-native-safe-area-context": "^4.7.2",
```
Comment (Cal-L) [inline package.json:304]: was this part of the upgrade?

File: package.json [hunk 4]
```diff
@@ -357,22 +359,30 @@
     "@metamask/eslint-config-typescript": "^9.0.0",
     "@metamask/mobile-provider": "^2.1.0",
     "@metamask/test-dapp": "^7.1.0",
+    "@react-native-community/datetimepicker": "^7.5.0",
     "@react-native-community/eslint-config": "^2.0.0",
+    "@react-native-community/slider": "^4.4.3",
```
Comment (Cal-L) [inline package.json:364]: Was this part of the upgrade?

File: package.json [hunk 5]
```diff
@@ -357,22 +359,30 @@
...
+    "@storybook/addon-ondevice-controls": "^6.5.6",
     "@storybook/addon-ondevice-knobs": "^5.3.25",
-    "@storybook/react-native": "^5.3.25",
+    "@storybook/addons": "^7.4.2",
+    "@storybook/core-common": "^7.4.2",
+    "@storybook/react-native": "^6.5.6",
     "@testing-library/react": "14.0.0",
     "@testing-library/react-hooks": "^8.0.1",
     "@testing-library/react-native": "12.1.2",
+    "@types/crypto-js": "^4.1.1",
```
Comment (Cal-L) [inline package.json:379]: Was this part of the upgrade?

File: package.json [hunk 6]
```diff
@@ -357,22 +359,30 @@
...
     "@types/enzyme": "^3.10.12",
     "@types/eth-url-parser": "^1.0.0",
     "@types/i18n-js": "^3.8.4",
     "@types/is-url": "^1.2.30",
     "@types/jest": "^28.1.8",
+    "@types/lodash": "^4.14.184",
```
Comment (Cal-L) [inline package.json:385]: Was this part of the upgrade?

**Review by Cal-L — COMMENTED** [review]
Left some comments

File: ios/Podfile.lock (hunk 1 above)
Comment (brianacnguyen) [reply ios/Podfile.lock:915]: they were part of the controls/knobs from the migration https://github.com/storybookjs/react-native/blob/v6.5/MIGRATION.md

File: .storybook/storybook.requires.js (hunk 2 above)
Comment (brianacnguyen) [reply .storybook/storybook.requires.js:15]: let me double check

File: package.json [hunk 7]
```diff
@@ -357,22 +359,30 @@
     "@metamask/eslint-config-typescript": "^9.0.0",
     "@metamask/mobile-provider": "^2.1.0",
     "@metamask/test-dapp": "^7.1.0",
+    "@react-native-community/datetimepicker": "^7.5.0",
```
[the remark this answers is not in the document]
Comment (brianacnguyen) [reply package.json:362]: yes https://github.com/storybookjs/react-native/blob/v6.5/MIGRATION.md

File: package.json (hunk 4 above)
Comment (brianacnguyen) [reply package.json:364]: yes https://github.com/storybookjs/react-native/blob/v6.5/MIGRATION.md

File: package.json (hunk 5 above)
Comment (brianacnguyen) [reply package.json:379]: will double check on this

File: package.json (hunk 6 above)
Comment (brianacnguyen) [reply package.json:385]: will double check on this

File: package.json (hunk 5 above)
Comment (brianacnguyen) [reply package.json:379]: this was simply moved from line 389

File: ios/Podfile.lock (hunk 1 above)
Comment (georgewrmarshall) [reply ios/Podfile.lock:915]: Does the `Podfile.lock` update if you are changing dependencies?

Comment (brianacnguyen) [reply ios/Podfile.lock:915]: I believe so

File: .storybook/storybook.requires.js (hunk 2 above)
Comment (brianacnguyen) [reply .storybook/storybook.requires.js:15]: we still have js stories so will keep this for now

File: package.json (hunk 3 above)
Comment (brianacnguyen) [reply package.json]: reverted this

File: ios/Podfile.lock [hunk 8]
```diff
@@ -345,7 +345,9 @@ PODS:
     - React-Core
   - react-native-randombytes (3.6.1):
     - React-Core
-  - react-native-safe-area-context (3.2.0):
+  - react-native-safe-area-context (3.4.1):
```
Comment (Cal-L) [inline ios/Podfile.lock:348]: What is causing this version to change? I suspect it was accidentally bumped in yarn lock?

Comment (brianacnguyen) [reply ios/Podfile.lock:348]: most likely. let me double check

**Review by Cal-L — APPROVED** [review]
LGTM

**Pull request merged**